core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushRemoteExchangeThroughAssignUniqueId.java

```diff
@@ -80,7 +80,8 @@ public Result apply(ExchangeNode node, Captures captures, Context context)
                                 removeSymbol(partitioningScheme.getOutputLayout(), assignUniqueId.getIdColumn()),
                                 partitioningScheme.getHashColumn(),
                                 partitioningScheme.isReplicateNullsAndAny(),
-                                partitioningScheme.getBucketToPartition()),
+                                partitioningScheme.getBucketToPartition(),
+                                partitioningScheme.getHashPartitionCount()),
```

Comment:
nit (can skip): you could add this extra new field in separate commit since it's mechanical and doesn't expose any new feature